lib/core/base/cache.js

```diff
@@ -13,10 +13,21 @@ const cache = {
   /**
    * Retrieve an item from the cache.
    * @param {String} key - Name of the key the value was stored as.
+   * @param {*} [defaultValue] - Default value to set if there is a cache miss. Functions are evaluated before caching. To override a value already saved, use `set()`.
    * @returns {*} The item stored
    */
-  get(key) {
-    return _cache[key];
+  get(key, defaultValue) {
+    if (key in _cache) {
+      return _cache[key];
+    }
+
+    if (arguments.length === 2) {
+      this.set(
+        key,
+        typeof defaultValue === 'function' ? defaultValue() : defaultValue
+      );
```

Comment:
I think we may need to throw if `creator` returns undefined. Because if it does we're just going to end up rerunning the creator method until it doesn't. That kind of defeats the purpose of the cache. Seeing how easy it is to get this wrong, I think it'd be safer not to allow undefined.

I spotted we had a bug because we're allowing this in isModalOpen (see my comments there).